projects/storefrontlib/src/events/page/page-event.builder.ts

```diff
@@ -17,7 +18,8 @@ import { HomePageEvent, PageEvent } from './page.events';
 export class PageEventBuilder {
   constructor(
     protected actions: ActionsSubject,
-    protected eventService: EventService
+    protected eventService: EventService,
+    protected contentPageMetaResolver: ContentPageMetaResolver
```

Comment:
I believe this isn't going to work, as the ContentPageMetaResolver is just one of the resolvers in the system. We have multiple standard resolvers for search, product, cart, but customers can also introduce specific resolvers themselves. 

Can you have a look at the `PageMetaService.getMeta()` method? That one will delegate to the right resolver to resolve the title. Also note that the title might not be resolved, but the `PageMeta` would still be given.